typescript/tsconfig.json

```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
```

Comment:
What can we do to guarantee that users, no matter what environment, will be able to import this project fine? Do we have to document, "please see our tsconfig settings"?

Would you expect the export/module TypeScript definition change to result in a new major rev (4.x.x)?